vendor/bat-native-confirmations/src/bat/confirmations/internal/confirmations_impl.cc

```diff
@@ -1055,4 +1055,22 @@ bool ConfirmationsImpl::IsRetryingToGetRefillSignedTokens() const {
   return true;
 }
 
+void ConfirmationsImpl::ResetConfirmationsState(
+    OnResetConfirmationsStateCallback callback) {
+  is_initialized_ = false;
+  state_has_loaded_ = false;
+  confirmations_client_->ResetState(_confirmations_name,
```

Comment:
this is what LedgerImpl should be doing - calling ResetState for the relevant state files